test/integration_test.go

```diff
@@ -2561,6 +2561,27 @@ func TestDependencyOutputTypeConversion(t *testing.T) {
 	assert.Equal(t, outputs["from_env"].Value, "default")
 }
 
+// Regression testing for https://github.com/gruntwork-io/terragrunt/issues/1102: Ordering keys from
+// maps to avoid random placements when terraform file if generated.
+func TestOrderedMapOutputRegressions1102(t *testing.T) {
+	t.Parallel()
+	generateTestCase := filepath.Join(TEST_FIXTURE_GET_OUTPUT, "regression-1102")
+
+	cleanupTerraformFolder(t, generateTestCase)
+	cleanupTerragruntFolder(t, generateTestCase)
+
+	stdout := bytes.Buffer{}
+	stderr := bytes.Buffer{}
+
+	runTerragruntCommand(t, fmt.Sprintf("terragrunt output -no-color -json --terragrunt-non-interactive --terragrunt-working-dir %s", generateTestCase), &stdout, &stderr)
+	expected, _ := ioutil.ReadFile(filepath.Join(generateTestCase, "backend.tf"))
+	for i := 1; i < 20; i++ {
+		runTerragruntCommand(t, fmt.Sprintf("terragrunt output -no-color -json --terragrunt-non-interactive --terragrunt-working-dir %s", generateTestCase), &stdout, &stderr)
+		actual, _ := ioutil.ReadFile(filepath.Join(generateTestCase, "backend.tf"))
```

Comment:
Check errors here instead of throwing them away (with `require.NoError`). Need to do this for the call above as well.

Reply:
Perfect, will change.